Framework/Algorithms/src/ConvertDiffCal.cpp

```diff
@@ -102,18 +105,18 @@ double calculateDIFC(OffsetsWorkspace_const_sptr offsetsWS,
 
   const detid_t detid = getDetID(offsetsWS, index);
   const double offset = getOffset(offsetsWS, detid);
-
-  double l1;
-  Kernel::V3D beamline, samplePos;
-  double beamline_norm;
-  instrument->getInstrumentParameters(l1, beamline, beamline_norm, samplePos);
-
-  Geometry::IDetector_const_sptr detector = instrument->getDetector(detid);
+  const Mantid::API::SpectrumInfo &spectrumInfo = offsetsWS->spectrumInfo();
```

Comment:
This seems to be done inside a loop. I think `spectrumInfo` can be passed in to this method instead of the workspace.